compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -299,23 +301,31 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         (caller_abi, caller_fn_abi): (Abi, &FnAbi<'tcx, Ty<'tcx>>),
         args: &[OpTy<'tcx, M::PointerTag>],
         with_caller_location: bool,
-        ret: Option<(&PlaceTy<'tcx, M::PointerTag>, mir::BasicBlock)>,
+        dest_place: &PlaceTy<'tcx, M::PointerTag>,
+        dest_block: Option<mir::BasicBlock>,
         mut unwind: StackPopUnwind,
     ) -> InterpResult<'tcx> {
         trace!("eval_fn_call: {:#?}", fn_val);
 
         let instance = match fn_val {
             FnVal::Instance(instance) => instance,
             FnVal::Other(extra) => {
-                return M::call_extra_fn(self, extra, caller_abi, args, ret, unwind);
+                return M::call_extra_fn(
+                    self,
+                    extra,
+                    caller_abi,
+                    args,
+                    dest_block.map(|x| (dest_place, x)),
```

Comment:
Looks like we should propagate this change through the machine interface all the way to Miri?

Reply:
Yeah, I had avoided doing that for now because I'm not sure what the right procedure for it is. Should I just make the change and then send a follow-up PR to miri?

Reply:
Yes. It is okay for Miri to temporarily break in the rustc repo.

Ideally you can prepare the Miri PR already before this one lands, so that once this one lands we can get Miri fixed ASAP.